### ISSUE TITLE
Sharing: the sharing settings section overall needs more detail about what the options and buttons are for

### ISSUE THREAD

**Issue by designsimply** [wordpress-mobile/WordPress-iOS#12890]
Steps to reproduce:

1. Go to Settings > General > Accessibility > VoiceOver and toggle the setting on.
1. In the WordPress app, go to My Site > Sharing.
1. Swipe right to navigate through each element.
1. Go to Sharing Buttons > Manage and try to add or remove a sharing button.
1. Go to Tumblr and try to connect a Tumblr account.

**Result:** [4m34s](https://cloudup.com/cKF7v3YcP5Z)

* Overall needs more detail about what the options and buttons are for.
* Error state is not communicated by Voiceover.
* Descriptions of the section should be read with the heading, not below all of the options.
* There is a lot of scroll bleed during the connection step where things in the screen below are read out and it's pretty confusing.

**[Accessibility Guidelines](https://developer.apple.com/documentation/uikit/accessibility/supporting_voiceover_in_your_app) that may help here:**

* Group elements so that VoiceOver navigates through them in the correct order. VoiceOver reads from the leading to the trailing edge. If you want VoiceOver to read your elements in a different order, use groups to facilitate navigation that makes sense for your app.
* Include descriptive text for VoiceOver to read. A UI that depends on visual cues may look nice, but can be unusable for a VoiceOver user. For example, VoiceOver doesn't detect if a confirmation button turns from gray to green when selected. VoiceOver may only describe the element and not it’s current state. Make sure that VoiceOver says whether you have selected the button.

<img src="https://user-images.githubusercontent.com/1119271/68325619-04d6ea80-0087-11ea-8dec-bf3c3dfc392c.png" width="200" alt="sharing-connections-error-icon"> <img src="https://user-images.githubusercontent.com/1119271/68325831-76169d80-0087-11ea-82cc-6bc14608c00b.png" width="200" alt="section-explanation-ordering"> <img src="https://user-images.githubusercontent.com/1119271/68325971-c3930a80-0087-11ea-9491-3bf2668a811c.png" width="200" alt="tumblr-connection-screen"> <img src="https://user-images.githubusercontent.com/1119271/68326127-15d42b80-0088-11ea-8d6e-27d39c4d74eb.png" width="200" alt="connection-permissions"> <img src="https://user-images.githubusercontent.com/1119271/68326258-4b791480-0088-11ea-8a52-0226637588e1.png" width="200" alt="connecting-tumblr">
<sup>Tested with WPiOS 13.5.0.3 (App Store version) on iPhone 6S iOS 13.2.</sup>

<p style="clear:both;"><sub>/hat tip @etoledom for the report, internal reference: p77Llu-bQ6-p2</sub></p>

**Comment by dangermattic** [comment]
Thanks for reporting! 👍
